core/src/test/java/com/linecorp/armeria/common/FlagsTest.java

```diff
@@ -107,10 +108,25 @@ void dumpOpenSslInfoDoNotThrowStackOverFlowError() throws Throwable {
 
         final MethodHandle dumpOpenSslInfoMethodHandle =
                 lookup.findStatic(flags, "dumpOpenSslInfo", MethodType.methodType(boolean.class));
-        // // Call Flags.dumpOpenSslInfo();
+        // Call Flags.dumpOpenSslInfo();
         assertThat(dumpOpenSslInfoMethodHandle.invoke()).isSameAs(Boolean.TRUE);
     }
 
+    @Test
+    void defaultTlsEngineType() {
+        assumeThat(System.getProperty("com.linecorp.armeria.tlsEngineType")).isNull();
+
+        assertThat(Flags.tlsEngineType()).isEqualTo(TlsEngineType.OPENSSL);
+    }
+
+    @Test
+    void tlsEngineTypeIsUsedWhenIncompatibleWithUseOpenSsl() {
+        System.setProperty("com.linecorp.armeria.useOpenSsl", "false");
+        System.setProperty("com.linecorp.armeria.tlsEngineType", "OPENSSL");
```

Comment:
I believe these values will be set while the jvm is running. Do you mind using either `assumeThat` or `@SetSystemProperty` instead?

```suggestion
        System.setProperty("com.linecorp.armeria.useOpenSsl", "false");
        System.setProperty("com.linecorp.armeria.tlsEngineType", "OPENSSL");
```

Reply:
I'll change the test to use `@SetSystemProperty`. 
cf) I think this test leads to the problems you [mentioned](https://github.com/line/armeria/pull/5029#discussion_r1304344623). Should we consider setting ways to prevent above test running in parallel(even if we use `--parallel` flag)